### PR TITLE
Publishing to IPNS should be in separate menu

### PR DESCRIPTION
**Pull request by spylogsster**
Resolves https://github.com/brave/brave-browser/issues/16998

image

## Submitter Checklist:

- [x] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [x] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [x] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

## Test Plan:

- Import items in the app menu should not be visible if local node is not selected as resolution method and not launched
- IPNS keys items should not be visible if ipns keys are empty in the service
- all import actions should work as expected

### CODE REVIEW

File: browser/ui/browser_commands.h [hunk 1]
```diff
@@ -21,7 +21,7 @@ void MaybeDistillAndShowSpeedreaderBubble(Browser* browser);
 void ShowBraveVPNBubble(Browser* browser);
 void ToggleBraveVPNButton(Browser* browser);
 void OpenBraveVPNUrls(Browser* browser, int command_id);
-
+void OpenIpfsFilesWebUI(Browser* browser);
```
Comment (yrliou) [inline browser/ui/browser_commands.h:24]: empty line between functions and namespaces.

File: browser/ui/toolbar/brave_app_menu_model.cc [hunk 2]
```diff
@@ -14,7 +14,7 @@
 #include "chrome/browser/profiles/profile.h"
 #include "chrome/grit/generated_resources.h"
 
-#if BUILDFLAG(ENABLE_IPFS)
+#if BUILDFLAG(ENABLE_IPFS_LOCAL_NODE)
```
Comment (yrliou) [inline browser/ui/toolbar/brave_app_menu_model.cc:17]: What's the reason for changing all flags from ENABLE_IPFS to ENABLE_IPFS_LOCAL_NODE in this PR?

Comment (spylogsster) [reply browser/ui/toolbar/brave_app_menu_model.cc:17]: All ipfs related code in this file works only when ipfs local node is up. We added ENABLE_IPFS_LOCAL_NODE later than this code and I changed it now to have proper build flag

File: browser/ui/browser_commands.h (hunk 1 above)
Comment (spylogsster) [reply browser/ui/browser_commands.h:24]: done

File: browser/ui/brave_browser_command_controller.cc [hunk 3]
```diff
@@ -39,7 +39,7 @@
 #include "brave/browser/ui/sidebar/sidebar_utils.h"
 #endif
 
-#if BUILDFLAG(ENABLE_IPFS)
+#if BUILDFLAG(ENABLE_IPFS_LOCAL_NODE)
 #include "brave/components/ipfs/ipfs_constants.h"
 #include "brave/components/ipfs/ipfs_utils.h"
 #include "brave/components/ipfs/pref_names.h"
```
[the remark this answers is not in the document]
Comment (spylogsster) [reply browser/ui/brave_browser_command_controller.cc]: done

File: app/brave_generated_resources.grd [hunk 4]
```diff
@@ -541,11 +541,23 @@ Or change later at <ph name="SETTINGS_EXTENIONS_LINK">$2<ex>brave://settings/ext
...
+        <message name="IDS_APP_MENU_IPFS_SHARE_LOCAL_FILE" desc="Title case: The text label of a menu item to share local file using IPFS.">
+          Import and Share a File
         </message>
-        <message name="IDS_APP_MENU_IPFS_IMPORT_LOCAL_FOLDER" desc="Title case: The text label of a menu item to share local fodler using IPFS.">
-          Share Local Folder Using IPFS
+        <message name="IDS_APP_MENU_IPFS_OPEN_FILES" desc="Title case: The text label of a menu item to open IPFS webui files page.">
+          Open Files
+        </message>
+        <message name="IDS_APP_MENU_IPFS_SHARE_LOCAL_FOLDER" desc="Title case: The text label of a menu item to share local fodler using IPFS.">
+          Import and Share a Directorys
```
Comment (mkarolin) [inline app/brave_generated_resources.grd:551]: Typo: Directory (no `s`)

**Review by yrliou — APPROVED** [review]

**Review by mkarolin — APPROVED** [review]
LGTM

**Pull request merged**